### PR TITLE
Add support for Kogan Smarter Home Plug With Energy Meter

### PR DESCRIPTION
**Pull request by aureq**
Type: Power Plug
Board: Kogan Smarter Home Plug with Energy Meter
Build flag: KOGAN_SMARTER_HOME_PLUG_W_POW
Description: Power Plug with integrated Energy Meter
Baseboard: Tuya TYWE3S
Baseboard Data sheet: https://docs.tuya.com/docDetail?code=K8uhkbb1ihp8u

### CODE REVIEW

**Comment by aureq** [comment]
Hi,

The work based in the PR is mostly based on the work done in #2010.

File: code/espurna/config/arduino.h [hunk 1]
```diff
@@ -96,6 +96,7 @@
 //#define JINVOO_VALVE_SM_AW713
 //#define JORGEGARCIA_WIFI_RELAYS
 //#define KMC_70011
+#define KOGAN_SMARTER_HOME_PLUG_W_POW
```
Comment (mcspr) [inline code/espurna/config/arduino.h]: Please comment this out.
*I wonder how CI test can catch this too. It already overrides device flag, so this does nothing there.*

File: code/espurna/config/hardware.h [hunk 2]
```diff
@@ -4211,6 +4211,54 @@
...
+// Reflashing from original Tuya firmware
+// to thirdparty firmware like espurna by:
+// https://github.com/ct-Open-Source/tuya-convert
+// -----------------------------------------------------------------------------
+
+#elif defined(KOGAN_SMARTER_HOME_PLUG_W_POW)
+
+    // Info
+    #define MANUFACTURER                "KOGAN"
+    #define DEVICE                      "SMARTER_HOME_PLUG_POW"
```
Comment (mcspr) [inline code/espurna/config/hardware.h]: `defined(...string....)` should match `MANUFACTURER` + `_` + `DEVICE`
We need to keep one style, either `PLUG_W_POW` or `PLUG_POW` at the end. Whichever you prefer.
ofc pio ini and arduino header should match

**Comment by aureq** [comment]
@mcspr Thank you for the feedback. I've pushed the code. Let me know if there's anything else I should do before you merge the changes. Thanks.

**Pull request merged**